### PR TITLE
Add bank filter by price range

### PR DESCRIPTION
**Pull request by tanlines**
#6778 

Option to filter by GE or HA price
Ability to set price range
Hides items in bank not within price range when doing any bank filtering, including with tags
When active, if no filter is being applied, automatically applies a "pricefilter" search to begin filtering items
    If user opens search box, this filter is disabled (bug: requires spam clicking to open search box if on All items tab)

Mostly due to my lack of knowledge of how the bank filtering works, I've spent the last couple days working on this plugin, hoping for some feedback especially on where I should be calling banksearch.search() so the bug doesn't occur.

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/banktags/BankTagsConfig.java [hunk 1]
```diff
@@ -53,6 +53,61 @@ default boolean rememberTab()
 		return true;
 	}
 
+	@ConfigItem(
```
Comment (raiyni) [inline runelite-client/src/main/java/net/runelite/client/plugins/banktags/BankTagsConfig.java]: This should just work if bank tags plugin is enabled

File: runelite-client/src/main/java/net/runelite/client/plugins/banktags/TagManager.java [hunk 2]
```diff
@@ -169,6 +178,35 @@ public void removeTag(int itemId, String tag)
 		}
 	}
 
+	public boolean itemWithinPriceRange(Item item)
```
Comment (raiyni) [inline runelite-client/src/main/java/net/runelite/client/plugins/banktags/TagManager.java]: Nothing is done with tag search, so why is this here?

File: runelite-client/src/main/java/net/runelite/client/plugins/banktags/BankTagsPlugin.java [hunk 3]
```diff
@@ -205,6 +206,21 @@ else if (!tagSearch)
 				{
 					intStack[intStackSize - 2] = itemName.contains(search) ? 1 : 0;
 				}
+
+				if (config.priceFilter()) {
+					if (search.equals(PRICE_FILTER)) {
+						intStack[intStackSize - 2] = 1;
+					}
+					if (intStack[intStackSize - 2] == 1) {
+						ItemContainer container = client.getItemContainer(InventoryID.BANK);
```
Comment (raiyni) [inline runelite-client/src/main/java/net/runelite/client/plugins/banktags/BankTagsPlugin.java]: This should be cached on itemcontainerchanged

File: runelite-client/src/main/java/net/runelite/client/plugins/banktags/TagManager.java (hunk 2 above)
Comment (tanlines) [reply runelite-client/src/main/java/net/runelite/client/plugins/banktags/TagManager.java]: I wasn't sure where to put it.
I considered the price of the item as a sort of "tag".

File: runelite-client/src/main/java/net/runelite/client/plugins/banktags/BankTagsConfig.java (hunk 1 above)
Comment (tanlines) [reply runelite-client/src/main/java/net/runelite/client/plugins/banktags/BankTagsConfig.java]: I don't think it should be on by default if bank tags is enabled, as it forces a filter on all items and all tags.

Comment (raiyni) [reply runelite-client/src/main/java/net/runelite/client/plugins/banktags/BankTagsConfig.java]: It shouldn't always be searching price value.

File: runelite-client/src/main/java/net/runelite/client/plugins/banktags/BankTagsPlugin.java (hunk 3 above)
Comment (tanlines) [reply runelite-client/src/main/java/net/runelite/client/plugins/banktags/BankTagsPlugin.java]: Are you saying I should cache it or it's already cached?
How does this change what I'm doing? I still need to check the items individually and filter them.

Comment (raiyni) [reply runelite-client/src/main/java/net/runelite/client/plugins/banktags/BankTagsPlugin.java]: You are querying the bank item container for every item in the bank and then looping over it.

File: runelite-client/src/main/java/net/runelite/client/plugins/banktags/BankTagsConfig.java (hunk 1 above)
Comment (tanlines) [reply runelite-client/src/main/java/net/runelite/client/plugins/banktags/BankTagsConfig.java]: The original issue didn't specify how to do the filtering.

How do you think it should be activated then?
If you have to type the keyword in every time the search resets it's very cumbersome in my opinion. Rather, I let the user toggle it on and off in the settings.

Comment (raiyni) [reply runelite-client/src/main/java/net/runelite/client/plugins/banktags/BankTagsConfig.java]: Through search tags of `ge` or `ha` would make more sense rather than always having a clunky filter on. 

Plus it's faster to enable it that way.

Comment (tanlines) [reply runelite-client/src/main/java/net/runelite/client/plugins/banktags/BankTagsConfig.java]: How about a "price:" tag, and let the user toggle HA or GE values in the sidebar? I don't think these would be changed very much, and easier to just remember 1 price tag.

It'll use a similar method of checking if the user is searching by tag and also let the user filter by item name as well.

Comment (raiyni) [reply runelite-client/src/main/java/net/runelite/client/plugins/banktags/BankTagsConfig.java]: why would you want that much complexity
